src/dex/balancer-v2/balancer-v2.ts

```diff
@@ -482,7 +565,7 @@ export class BalancerV2
               _from,
               _to,
               pool,
-              poolState,
+              allPoolStates[poolAddress] ? allPoolStates : missingPoolsStateMap,
```

Comment:
shouldn't you merge the `allPoolStates` and `missingPoolsStateMap`? what if some base pools are there is one of the object while  the others are there in other?